### DISCUSSION TITLE
Licensing...?

### DISCUSSION THREAD

**Discussion by wowogiengen** [PredatH0r/ChanSort#278] [Q&A]
Hello, I am very pleased that i found this channel list editor. It worked on my channel list of a 6 year old Panasonic DMR-BST845 BluRay and HDD recorder. 
I am a software developper myself, and wanted to compile the given source code. But it seems like I need a full DevExpress License to do so. 
This is a quite expensive task and i dont think that many hobbyist programmers can afford such a license.

Why did you choose to use these UI elements? And is there a way to circumvent the compiler errors because of the license?

**Comment by PredatH0r** [comment]
You can select between different Build configurations in Visual Studio. Besides the normal Debug and Release, there are 2 more in the ChanSort solution that skip building the UI. That way you can build your own TV loader modules and just use the precompiled UI DLL/EXE.

I have a paid DevExpress license and over a decade of experience with them. Especially the data grid is so much more powerful than anything the standard .NET WinForms controls can offer, that I didn't want to torture myself by using something I am not familiar with and that's a lot more effort to achieve the same task.

Back then DevExpress was less restrictive and offered a demo-version download of their controls. Only downside was that you got a nag-screen when you started compiling and the .exe, but that was easy to click away.

Here you can find some information on how to build ChanSort yourself:
https://github.com/PredatH0r/ChanSort/blob/master/source/build.md
